### ISSUE TITLE
Get an error saying cannot import next/document outside of _document even though it is in _document

### ISSUE THREAD

**Issue by callumok2004** [vercel/next.js#29021]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

16.6.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

When running `next lint` I get the following output.

```
yarn run v1.22.10
$ next lint --fix
info  - Loaded env from E:\Developing\Ansen\Recode\.env
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5

./src/pages/_document.tsx
1:1  Error: next/document should not be imported outside of pages/_document.js. See https://nextjs.org/docs/messages/no-document-import-in-page.  @next/next/no-document-import-in-page

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
```

### Expected Behavior

It should not be giving the error as the import IS inside _document

### To Reproduce

Create a `_document.tsx` file in `src/pages/`, have an import for `next/document`, e.g `import Document, { Head, Html, Main, NextScript } from 'next/document';`

run `next lint`

**Comment by nbouvrette** [comment]
I was about to open this issue - got the same problem :)

**Comment by junbach** [comment]
I got exactly same issue

**Comment by gabor-at-reed** [comment]
Same here

**Comment by andrei9669** [comment]
seems like a regression issue [here](https://github.com/vercel/next.js/blob/canary/packages/eslint-plugin-next/lib/rules/no-document-import-in-page.js) since it's working in 11.1.0

**Comment by housseindjirdeh** [comment]
This is likely related to #28596. Is `eslint-config-next` (or `@next/eslint-plugin-next`) installed with version `11.1.3-canary.7` or later? If not, upgrading it should fix this issue.

**Comment by daliudzius** [comment]
Same issue, I have pages/_document.js under the alternative setup of src/pages/_document.js -- maybe this is the reason for my false positive error.

**Comment by housseindjirdeh** [comment]
@daliudzius What version of `eslint-config-next` or `@next/eslint-plugin-next` are you using?

**Comment by daliudzius** [comment]
@housseindjirdeh My `eslint-config-next` is on 11.1.2

**Comment by ijjk** [comment]
Closing as a duplicate of https://github.com/vercel/next.js/issues/28596 @daliudzius the fix is in `11.1.3-canary.7` please update to at least that version to resolve this.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
